HHS/Head-Start-TTADP
# Fix migrations/Add URL routing to activity report

# Issue 202
**Description of change**
- adds migration to deployment command
- cleans up other environment variables that are passed to the application
- adds step by step instructions for developers to add their own env variables to the application.

**How to test**
- Add staging postgres db vars to circle ci as env variable. @rahearn Use these variable names: STAGING_POSTGRES_DB, STAGING_POSTGRES_HOST, STAGING_POSTGRES_PASSWORD, STAGING_POSTGRES_USERNAME
- After deploy completes, check that the postgres var were written as cloud.gov env vars and verify that the migrations were run by reading the log stream. Failed migrations will not be visible in circle ci since the migration is run as a task on the application instance

**Issue(s)**
* closes https://github.com/HHS/Head-Start-TTADP/issues/202

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

---

# Issue 135
**Description of change**
The current activity reports page shown is determined by the url. Being able to browse to specific report pages via URL has two main advantages:
 * A user can share/bookmark a specific page
 * We are setup to directly link to specific sections of the pages

Also I added the report pages to the axe testing, and added a file with the urls axe should test. Note the `axe:ci` command no longer works on windows. Also fixed a color contrast accessibility issue in the activity report side nav when a page is selected and has a state of `Complete`.

**How to test**

Browse to [dev](https://tta-smarthub-dev.app.cloud.gov/activity-reports). Browsing to different pages will change the url. You can also directly go to specific pages of the report (https://tta-smarthub-dev.app.cloud.gov/activity-reports/review)

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/135

**Checklist**

- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

---

# Issue 230
**Description of change**
Updates nodejs version to `12.20.0` to work with current cloud.gov buildpack

**Issue**
* closes https://github.com/HHS/Head-Start-TTADP/issues/230

**Checklist**
**Checklist**
